File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
    unordered_map<char,int> m;
    if(s.size()!=t.size()){
        return false;
    }
    for(char c:s){
        m[c]++;
    }
    for(char c:t){
        if(m.find(c)==m.end()){
            return false;

        }
        else{
            if(m[c]==0){
                return false;
            }
            else{
                m[c]--;
            }
        }
    }
    return true;
        
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int m=s.length();
        int n=t.length();
        if(n!=m){
            return false;
        }
        HashMap<Character,Integer> map=new HashMap<>();
        for(int i=0;i<m;i++){
            char ch=s.charAt(i);
            map.put(ch,map.getOrDefault(ch,0)+1);
        }
        for(int i=0;i<n;i++){
            if(!map.containsKey(t.charAt(i))){
                return false;
            }
            if(map.containsKey(t.charAt(i))){
                map.put(t.charAt(i),map.get(t.charAt(i))-1);
            }
            if(map.get(t.charAt(i))==0){
                map.remove(t.charAt(i));
            }
        }
        return map.isEmpty();
    }
}